src/tests/hash_functions/validation/generate_hash_arrays.cpp

```diff
@@ -0,0 +1,174 @@
+#include <iostream>
+#include <fstream>
+
+extern "C" {
+  #include "nmhash.h"
+  #include "nmhash_scalar.h"
+  #include "pengyhash.h"
+  #include "waterhash.h"
+}
+
+#include "SpookyV2.h"
+
+void SpookyHash32_with_state_test(const void *key, size_t len, const void *state, void *out) {
+  uint64_t *state64= (uint64_t *)state;
+  uint64_t s0 = state64[0];
+  uint64_t s1 = state64[1];
+  SpookyHash::Hash128(key, len, &s0, &s1);
+  ((uint32_t *)out)[0]= (uint32_t)s0;
+}
+
+void SpookyHash64_with_state_test(const void *key, size_t len, const void *state, void *out) {
+  uint64_t *state64= (uint64_t *)state;
+  uint64_t *out64= (uint64_t *)out;
+  out64[0] = state64[0];
+  uint64_t s1 = state64[1];
+  SpookyHash::Hash128(key, len, out64, &s1);
+}
+
+void SpookyHash128_with_state_test(const void *key, size_t len, const void *state, void *out) {
+  uint64_t *state64= (uint64_t *)state;
+  uint64_t *out64= (uint64_t *)out;
+  out64[0] = state64[0];
+  out64[1] = state64[1];
+  SpookyHash::Hash128(key, len, out64, out64+1);
+}
+
+void SpookyHash_seed_state_test(int in_bits, const void *seed, void *state) {
+    uint64_t *state64= (uint64_t *)state;
+    if (in_bits == 32) {
+        state64[0]= state64[1]= ((uint32_t*)seed)[0];
+    }
+    else {
+        uint64_t *seed64= (uint64_t *)seed;
+        if (in_bits == 64) {
+            state64[0]= state64[1]= seed64[0];
+        }
+        else
+        if (in_bits == 128) {
+            state64[0]= seed64[0];
+            state64[1]= seed64[1];
+        }
+    }
+}
+
+using namespace std;
```

Comment:
I think it's considered bad practice to import std. If I'm not mistaken it's needed only for std::string?

Reply:
Maybe, but I am far from a C++ expert and when I try changing

    using namespace std;

to

    using namespace std::string

I get an error cascade starting with "error: 'string' is not a namespace-name". Any guidance would be appreciated.